src/index.test.ts

```diff
@@ -1425,6 +1425,91 @@ describe('TypedDataUtils.encodeData', function () {
   });
 });
 
+describe('TypedDataUtils.encodeType', () => {
+  it('should encode simple type', () => {
+    const types = {
+      Person: [{ name: 'name', type: 'string' }],
+    };
+    const primaryType = 'Person';
+
+    expect(
+      sigUtil.TypedDataUtils.encodeType(primaryType, types),
+    ).toMatchInlineSnapshot(`"Person(string name)"`);
+  });
+
+  it('should encode complex type', () => {
+    const types = {
+      Person: [
+        { name: 'name', type: 'string' },
+        { name: 'wallet', type: 'address' },
+      ],
+      Mail: [
+        { name: 'from', type: 'Person' },
+        { name: 'to', type: 'Person[]' },
+        { name: 'contents', type: 'string' },
+      ],
+    };
+    const primaryType = 'Mail';
+
+    expect(
+      sigUtil.TypedDataUtils.encodeType(primaryType, types),
+    ).toMatchInlineSnapshot(
+      `"Mail(Person from,Person[] to,string contents)Person(string name,address wallet)"`,
+    );
+  });
+
+  it('should encode recursive type', () => {
+    const types = {
+      Person: [
+        { name: 'name', type: 'string' },
+        { name: 'wallet', type: 'address' },
+      ],
+      Mail: [
+        { name: 'from', type: 'Person' },
+        { name: 'to', type: 'Person' },
+        { name: 'contents', type: 'string' },
+        { name: 'replyTo', type: 'Mail' },
+      ],
+    };
+    const primaryType = 'Mail';
+
+    expect(
+      sigUtil.TypedDataUtils.encodeType(primaryType, types),
+    ).toMatchInlineSnapshot(
+      `"Mail(Person from,Person to,string contents,Mail replyTo)Person(string name,address wallet)"`,
+    );
+  });
+
+  it('should encode unrecognized types', () => {
```

Comment:
I did a double-take reading this and the missing primary type test case, and we can spare future readers the trouble:

```suggestion
  it('should encode unrecognized non-primary types', () => {
```